src/logic/otp2/controller.ts

```diff
@@ -536,15 +525,136 @@ async function searchTaxiFrontBack(
     }
 }
 
-export type NonTransitMode = 'foot' | 'bicycle' | 'car' | 'bike_rental'
+/**
+ * TODO 7/12-21: Clean up function signature and types once OTP1 is gone.
+ * As we have removed recursion, initialSearchDate can be replaced with
+ * searchDate. searchFilter could
+ * be kept separate from Otp2SearchParams.
+ */
+export async function searchTransit(
+    { initialSearchDate, searchFilter, ...searchParams }: Otp2SearchParams,
+    extraHeaders: { [key: string]: string },
+): Promise<TransitTripPatterns> {
+    const { searchDate, arriveBy } = searchParams
+
+    const getTripPatternsParams = {
+        ...searchParams,
+        modes: filterModesAndSubModes(searchFilter),
+    }
+
+    // We do two searches in parallel here to speed things up a bit. One is a
+    // flexible search where we explicitly look for trips that may include means
+    // of transport that has to be booked in advance the second is a regular
+    // search.
+    const [
+        flexibleResults,
+        [regularTripPatternsUnfiltered, initialMetadata, routingErrors],
+    ] = await Promise.all([
+        searchFlexible(searchParams),
+        getTripPatterns(getTripPatternsParams),
+    ])
+
+    let queries = [
+        ...flexibleResults.queries,
+        getTripPatternsQuery(getTripPatternsParams),
+    ]
+
+    let metadata = initialMetadata
+
+    let regularTripPatterns = regularTripPatternsUnfiltered
+        .filter(isValidTransitAlternative)
```

Comment:
@draperunner what would be the consequence of moving the filtering into getTripPatterns? Would it change anything for flexible, taxi or non transit? I found one place where we (I?) forgot the filtering...

Reply:
I think `isValidTransitAlternative`'s mot important job is to check that a trip pattern actually includes a transit leg. But it has some legacy check on flexible stuff... I'd keep it as-is for now, and then we can review our filtering logic when removing OTP 1.